### PR TITLE
feat: consul deduplicate and ip port sort

### LINKED ISSUE #10882: feat: As a user, I want the consul plug-in to call the serviceID of the health api for deduplication.
### Description

According to [member blake of consul](https://github.com/hashicorp/consul/issues/11438#issuecomment-1003459485), the best way to implement consul is to register it on each node, but apisix starts from the api: v1/health/checks/ takes serviceID to get the data of each node, so there will be duplicate data here, for example
``` json
[{"Node":"consul-01","CheckID":"service:192.168.11.2:8001","Name":"Service 'test' check","Status":"passing","Notes": "","Output":"TCP connect 192.168.11.2:8001: Success","ServiceID":"192.168.11.2:8001","ServiceName":"test","ServiceTags":[],"Type": "tcp","Interval":"5s","Timeout":"5s","ExposedPort":0,"Definition":{},"CreateIndex":12014,"ModifyIndex":12015},{"Node" :"consul-03","CheckID":"service:192.168.11.2:8001","Name":"Service 'test' check","Status":"passing","Notes":"","Output ":"TCP connect 192.168.11.2:8001: Success","ServiceID":"192.168.11.2:8001","ServiceName":"test","ServiceTags":[],"Type":"tcp"," Interval":"5s","Timeout":"5s","ExposedPort":0,"Definition":{},"CreateIndex":10366,"ModifyIndex":12020}]
```
If node A is registered but node B is not, then the serviceID obtained by apisix will not be balanced, and the weight of the serviceID will be offset, but the user does not know it.
So, I want to deduplicate serviceID

By the way, I want to add a scheduling based on the minimum IP load. I still use roundrobin, but when polling the upstream, the polling is based on the minimum IP schedule, because we currently encounter a scenario where one machine has multiple service, but at a certain point in time, many requests hit the machine at the same time, causing the CPU to skyrocket. We hope to design an optional plug-in parameter. If this parameter is selected, the IP will be sorted first, and then the port will be sorted.

### PR DESCRIPTION
**Pull request by theweakgod**
### Description

Fixes #10882 

Remove duplicate services caused by each node in the Consul cluster, and add sorting by host_sort, port_sort, combine_sort.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by monkeyDluffy6017** [comment]
please make the ci pass

**Comment by monkeyDluffy6017** [comment]
Can we add test cases to cover this?

**Comment by theweakgod** [comment]
> Can we add test cases to cover this?

yeah, I add it, But I forgot it😂.

**Comment by theweakgod** [comment]
@monkeyDluffy6017 It's already been added.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by Neilblaze — APPROVED** [review]

File: t/discovery/consul.t [hunk 1]
```diff
@@ -781,3 +781,179 @@ location /sleep {
     qr//
 ]
 --- ignore_error_log
+
```
Comment (monkeyDluffy6017) [inline t/discovery/consul.t:784]: This file is too long, you should create a new test file

Comment (theweakgod) [reply t/discovery/consul.t]: done

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by monkeyDluffy6017 — APPROVED** [review]

**Pull request merged**